README.md

```diff
@@ -15,7 +94,52 @@ have pre-slotted error codes for each library. The currently allocated error ran
 | [0, 0x0400) | aws-c-common |
 | [0x0400, 0x0800) | aws-c-io |
 | [0x0800, 0x1000) | aws-c-http |
-| [0x1000, 0x2000) |aws-c-amazon-flow | 
+| [0x1000, 0x2000) |aws-c-eventstream | 
 | [0x2000, 0x2800) | (reserved for future project) |
 
-Each library should begin its error codes at the beginning of its range and follow in sequence (don't skip codes).
+Each library should begin its error codes at the beginning of its range and follow in sequence (don't skip codes). Upon
+adding an AWS maintained library, an error code range must be approved and added to the above table.
+
+### Testing
+PRs without unit tests will not be acknowledged. A couple of tips:
+
+#### Aws Test Harness
+We provide a test harness for writing unit tests. This includes an allocator that will fail your test if you have any 
+memory leaks, as well as some `ASSERT` macros. To write a test:
+
+* Create a *.c test file in the tests directory of the project. 
+* Implement one or more tests with the signature `int test_case_name(struct aws_allocator *, void *ctx)`
+* Use the `AWS_TEST_CASE` macro to declare the test.
+* Include your test in the `tests/main.c` file.
+* Include yur test in the `tests/CMakeLists.txt` file. 
+
+### Coding Style
```

Comment:
Do we have any automated coding style enforcement?

Reply:
I'll add cpp style to the travis runs at some point.

cppcheck also does some of this. Marco added this recently but we backed it out because we had so much in flight. We can add it back shortly, since the churn has turned down a bit.